quinn-proto/src/connection/mod.rs

```diff
@@ -424,14 +448,23 @@ where
         };
 
         let mut buf = Vec::with_capacity(self.mtu as usize);
+        // Reserving capacity can provide more capacity than we asked for.
+        // However we are not allowed to write more than MTU size. Therefore
+        // the maximum capacity is tracked separately.
```

Comment:
This sounds like a bug in the existing code. Worth pulling out into a separate PR so we can merge ASAP?

Reply:
Sounds reasonable. Moved it into #954. Also the "cmsg" encoding bug. It doesn't hurt without this change, since only a single cmsg is encoded. But it also won't hurt to treat it independently from this feature.